sasdata/fair_database/data/test/test_datafile.py

```diff
@@ -0,0 +1,380 @@
+            username="testUser", password="secret", id=2
+        )
+        cls.private_test_data = DataFile.objects.create(
+            id=3, current_user=cls.user, file_name="cyl_400_20.txt", is_public=False
+        )
+        cls.private_test_data.file.save(
+            "cyl_400_20.txt", open(find("cyl_400_20.txt"), "rb")
+        )
+        cls.client1 = APIClient()
+        cls.client1.force_authenticate(user=cls.user)
+
+    # Test list public data
+    def test_does_list_public(self):
+        request = self.client1.get("/v1/data/file/")
+        self.assertEqual(
+            request.data,
+            {"public_data_ids": {1: "cyl_400_40.txt", 3: "cyl_400_20.txt"}},
+        )
+
+    # Test list a user's private data
+    def test_does_list_user(self):
+        request = self.client1.get(
+            "/v1/data/file/", data={"username": "testUser"}, user=self.user
+        )
+        self.assertEqual(request.data, {"user_data_ids": {3: "cyl_400_20.txt"}})
+
+    # Test list another user's public data
+    def test_list_other_user(self):
+        client2 = APIClient()
+        request = client2.get(
+            "/v1/data/file/", data={"username": "testUser"}, user=self.user
+        )
+        self.assertEqual(request.data, {"user_data_ids": {}})
+
+    # Test list a nonexistent user's data
+    def test_list_nonexistent_user(self):
+        request = self.client1.get("/v1/data/file/", data={"username": "fakeUser"})
+        self.assertEqual(request.status_code, status.HTTP_404_NOT_FOUND)
+
+    # Test loading a public data file
+    def test_does_load_data_info_public(self):
+        request = self.client1.get("/v1/data/file/1/")
+        self.assertEqual(request.status_code, status.HTTP_200_OK)
+
+    # Test loading private data with authorization
+    def test_does_load_data_info_private(self):
+        request = self.client1.get("/v1/data/file/3/")
+        self.assertEqual(request.status_code, status.HTTP_200_OK)
+
+    # Test loading data that does not exist
+    def test_load_data_info_nonexistent(self):
+        request = self.client1.get("/v1/data/file/5/")
+        self.assertEqual(request.status_code, status.HTTP_404_NOT_FOUND)
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.public_test_data.delete()
+        cls.private_test_data.delete()
+        cls.user.delete()
+        shutil.rmtree(settings.MEDIA_ROOT)
+
+
+class TestingDatabase(APITestCase):
+    """Test non-get methods for DataFile."""
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.user = User.objects.create_user(
+            username="testUser", password="secret", id=1
+        )
+        cls.data = DataFile.objects.create(
+            id=1, current_user=cls.user, file_name="cyl_400_20.txt", is_public=False
+        )
+        cls.data.file.save("cyl_400_20.txt", open(find("cyl_400_20.txt"), "rb"))
+        cls.client1 = APIClient()
+        cls.client1.force_authenticate(user=cls.user)
+        cls.client2 = APIClient()
```

Comment:
The tests look great. I like that they are small and that the names show what they test. Here I would suggest renaming `client1` to `client_authenticated` and `client2` to `client_unauthenticated` to make the intent clear.